command-line-downloader/src/main/java/uk/ac/ebi/ena/app/utils/MenuUtils.java

```diff
@@ -109,28 +116,73 @@ public static boolean isValidEmailAddress(String email) {
         return result;
     }
 
-    public static Map<String, List<String>>  parseAccessions(String accessions) {
+    public static Map<String, List<String>> parseAccessions(String accessions) {
         Map<String, List<String>> accessionDetailsMap = new HashMap<>();
         if (StringUtils.isEmpty(accessions)) {
 
             return accessionDetailsMap;
         } else if (new File(accessions).exists()) {
-            List<String> accessionList =  accsFromFile(accessions);
+            List<String> accessionList = accsFromFile(accessions);
 
-            return CommonUtils.processAccessions(accessionList);
+            return CommonUtils.processAccessions(Objects.requireNonNull(accessionList));
         } else {
             List<String> accessionList = Arrays.asList(accessions.split(","));
 
             return CommonUtils.processAccessions(accessionList);
         }
     }
 
-    @SneakyThrows
     public static List<String> accsFromFile(String inputValues) {
-        Stream<String> lines = Files.lines(Paths.get(inputValues), StandardCharsets.US_ASCII);
+        try {
+            String[] headers = Files.lines(Paths.get(inputValues))
+                    .map(s -> s.split("\t"))
+                    .findFirst()
+                    .orElse(new String[1]);
+
+            String firstColumn = headers[0];
+
+            if (firstColumn == null) {
+                return new ArrayList<>();
+            } else {
+                if (headers.length > 1) {
+                    TsvParserSettings settings = new TsvParserSettings();
+                    settings.getFormat().setLineSeparator("\n");
+                    settings.selectFields(firstColumn);
+
+                    TsvParser parser = new TsvParser(settings);
+                    List<String[]> accessionIdColumn = parser.parseAll(new File(inputValues));
+                    //removing the column header
+                    if (firstColumn.contains(ACCESSION)) {
+                        accessionIdColumn = accessionIdColumn.subList(1, accessionIdColumn.size());
+                    }
+
+                    return accessionIdColumn.stream().map(aRow -> aRow[0].replace("\"", "").trim())
+                            .collect(Collectors.toList());
+                } else {
+                    //if multiple headers are not present then we expect the file to contain only the accessionIds
```

Comment:
Sorry, I see I didn't mention this in the chat. This can also be a valid input file. i.e. no header, multiple columns. Only look at the first column and ignore the rest.

ERR4082025	ftp.sra.ebi.ac.uk/vol1/fastq/ERR408/005/ERR4082025/ERR4082025_1.fastq.gz	91988098
ERR4082026	ftp.sra.ebi.ac.uk/vol1/fastq/ERR408/006/ERR4082026/ERR4082026_1.fastq.gz	186032909
ERR4082027	ftp.sra.ebi.ac.uk/vol1/fastq/ERR408/007/ERR4082027/ERR4082027_1.fastq.gz	162622331
ERR4082028	ftp.sra.ebi.ac.uk/vol1/fastq/ERR408/008/ERR4082028/ERR4082028_1.fastq.gz	49719348
ERR4082029	ftp.sra.ebi.ac.uk/vol1/fastq/ERR408/009/ERR4082029/ERR4082029_1.fastq.gz	117361407

Reply:
Let me know if I am missing something